specification/profiling.md

```diff
@@ -0,0 +1,169 @@
+  MUST be set to the constant value `otel.profiling`.
+* [Time](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-timestamp)
+  MUST be set to the time that the call stack was sampled.
+* [TraceId](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-traceid)
+  MUST be populated when a call stack has been sampled within a span scope.
+* [SpanId](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-spanid)
+  MUST be populated when a call stack has been sampled within a span scope.
+* [Body](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-body)
+  MUST be populated with the line-terminated call stack (see below)
+
+##### Call Stack Format
+
+The call stack is a series of lines separated by newlines (`\n`).
+The first line of the call stack is a REQUIRED thread metadata line.
+An example metadata is shown here:
+```
+"pool-1-thread-30" #49 prio=5 os_prio=31 cpu=0.12ms elapsed=8.50s tid=0x00007fde4d00c000 nid=0xba03 waiting on condition  [0x000070000ba40000]
+```
+The metadata line consists of several space separated fields.
+* thread name, surrounded with double quotes `"`
+* `#<n>` index of this thread
+* `prio=<n>` where `<n>` is the thread priority. 0 if not available.
+* `os_prio=<n>` where `<n>` is the operating system thread priority. 0 if not available.
+* `cpu=<n><u>` where `<n>` is the amount of cpu time consumed by the thread and `<u>` is a time unit abbreviation. 0 if not available.
+* `elapsed=<n><u>` where `<n>` is the wall time that thread has been alive and `<u>` is a time unit abbreviation. 0 if not available.
+* `tid=0x<n>` where `<n>` is the lowercase hexadecimal representation of the thread id. 0 if not available.
+* `nid=0x<n>` where `<n>` is the lowercase hexadecimal representation of the native thread id. 0 if not available.
+* The remainder of the line is a free-form text field that SHOULD indicate the status of the thread at the time the stack was captured (such as "running" or "waiting" )
+
+Following the metadata line is a line containing the implementation specific thread state.
+For example, java may provide:
+
+```
+   java.lang.Thread.State: TIMED_WAITING (sleeping)
+```
+The thread states for other runtimes are currently undefined.
+
+
+The call stack details follow the thread state line. The call stack MUST have the top of the stack
+provided first and the bottom of the stack provided last. The format of each stack trace
+line is:
+
+```
+        at <class>.<method>(<source-module>:<line>)
+```
+
+* OPTIONAL leading whitespace
+* OPTIONAL "at "
+* `<class>` - REQUIRED. Fully scoped name of the class or module. If the function being executed is globally scoped this MUST be
```

Comment:
Not all languages are object oriented :) although, typically there are equivalent constructs (packages, modules, etc). Related to the next command: it should be the function/method signature according to the language.

Reply:
+1

pprof uses notions of `function`, `file` and `line`. For Java `function` can be represented as a method signature. Other languages may have other ways of presenting the signature of a function.

Reply:
[OTel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#source-code-attributes) uses `namespace`, `function`, `lineno` - we could use similar/same names here for consistency.

Reply:
This is great info folks, thank you. 

I like the idea of sticking with otel terms as @mateuszrzeszutek suggested: `namespace`, `function`, `lineno`.  Will also change `source-module` to `file` as @vovencij suggested.  👍🏻 ty